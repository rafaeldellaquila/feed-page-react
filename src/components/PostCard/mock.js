const postCardMock = {
  title: 'sunt aut facere repellat provident',
  body: 'quia et suscipit suscipit recusandae consequuntur',
  cover: 'https://via.placeholder.com/600/92c952',
  id: 0
}

export default postCardMock
