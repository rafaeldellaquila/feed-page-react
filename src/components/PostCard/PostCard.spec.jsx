import { render, screen } from '@testing-library/react'
import PostCard from '.'
import postCardMock from './mock'

describe('<PostCard/>', () => {
  it('should render the PostCard', () => {
    render(<PostCard {...postCardMock} />)

    expect(screen.getByRole('img', { name: postCardMock.title })).toHaveAttribute('src', postCardMock.cover)

    expect(screen.getByRole('heading', { name: postCardMock.title })).toBeInTheDocument()

    expect(screen.getByText(postCardMock.body)).toBeInTheDocument()
  })
  it('should match snapshot', () => {
    const { container } = render(<PostCard {...postCardMock} />)
    expect(container).toMatchSnapshot()
  })
})
