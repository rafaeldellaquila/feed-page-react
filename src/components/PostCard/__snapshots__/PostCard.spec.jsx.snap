// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div>
  <div
    class="post"
    id="0"
  >
    <img
      alt="sunt aut facere repellat provident"
      src="https://via.placeholder.com/600/92c952"
    />
    <div
      class="post-content"
    >
      <h2>
        sunt aut facere repellat provident
      </h2>
      <p>
        quia et suscipit suscipit recusandae consequuntur
      </p>
    </div>
  </div>
</div>
`;
